internal/pkg/expanders/storageBlob.go

```diff
@@ -0,0 +1,682 @@
+		ctx,
+		currentItem,
+		func(currentItem *TreeNode, blob Blob) (*TreeNode, error) {
+			node := TreeNode{
+				Parentid:  currentItem.ID,
+				Namespace: "storageBlob",
+				ID:        currentItem.ID + "/" + blob.Name,
+				Name:      blob.Name,
+				Display:   blob.Name,
+				ItemType:  storageBlobNodeBlob,
+				ExpandURL: ExpandURLNotSupported,
+				DeleteURL: currentItem.ID + "/" + blob.Name,
+				Metadata: map[string]string{
+					"BlobName": blob.Name,
+				},
+			}
+
+			return &node, nil
+		},
+		storageBlobNodeListBlob)
+}
+
+func (e *StorageBlobExpander) expandList(ctx context.Context, currentItem *TreeNode, createNodeFunc func(currentItem *TreeNode, blob Blob) (*TreeNode, error), continuationItemType string) ExpanderResult {
+
+	// https://docs.microsoft.com/en-us/rest/api/storageservices/enumerating-blob-resources#Subheading5
+
+	containerID := currentItem.Metadata["ContainerID"]
+	containerName := e.getContainerName(containerID)
+	accountName := e.getAccountName(containerID)
+	accountKey, err := e.getAccountKey(ctx, containerID)
+	marker := currentItem.Metadata["Marker"]
+	if err != nil {
+		err = fmt.Errorf("Error getting account key: %s", err)
+		return ExpanderResult{
+			Err:               err,
+			SourceDescription: "StorageBlobExpander request",
+		}
+	}
+	blobEndpoint, err := e.getStorageBlobEndpoint(ctx, containerID)
+	if err != nil {
+		err = fmt.Errorf("Error getting blob endpoint: %s", err)
+		return ExpanderResult{
+			Err:               err,
+			SourceDescription: "StorageBlobExpander request",
+		}
+	}
+
+	// ListBlob docs: https://docs.microsoft.com/en-us/rest/api/storageservices/list-blobs
+	url := blobEndpoint + containerName + "?restype=container&comp=list&maxresults=50"
+	if marker != "" {
+		url += "&marker=" + marker
+	}
+	buf, err := e.doRequest(ctx, "GET", url, accountName, accountKey, "/"+accountName+"/"+containerName)
+
+	if err != nil {
+		return ExpanderResult{
+			Err:               fmt.Errorf("Error listing blobs: %s", err),
+			SourceDescription: "StorageBlobExpander request",
+		}
+	}
+
+	response := &ContainerListResponse{}
+	err = xml.Unmarshal(buf, response)
+	if err != nil {
+		return ExpanderResult{
+			Err:               fmt.Errorf("Error Unmarshalling ContainerListResponse: %s", err),
+			SourceDescription: "StorageBlobExpander request",
+		}
+	}
+	nodes := []*TreeNode{}
+
+	for _, blob := range response.Blobs {
+		node, err := createNodeFunc(currentItem, blob)
+
+		if err != nil {
+			return ExpanderResult{
+				Err:               err,
+				SourceDescription: "StorageBlobExpander request",
+			}
+		}
+		node.Metadata["ContainerID"] = containerID
+		node.Metadata["AccountName"] = accountName
+		node.Metadata["AccountKey"] = accountKey
+
+		nodes = append(nodes, node)
+	}
+	if response.NextMarker != "" {
+		node := TreeNode{
+			Parentid:  currentItem.ID,
+			Namespace: "storageBlob",
+			ID:        currentItem.ID + "/" + "...more",
+			Name:      "more...",
+			Display:   "more...",
```

Comment:
oh this is a very cool solution

Reply:
nit: My expectation was to see the new items added to the existing list rather than stack a new page. Practically speaking tho I think this would be hard with the way expanders are built at the moment and this is a nice way to get the functionality in.
![image](https://user-images.githubusercontent.com/1939288/97812712-11caf180-1c7b-11eb-992c-2f95fd34060c.png)

Reply:
There's a precedent for the current approach in the container registry expander. I agree that it would be nice to extend the existing list and added #282